common/lib/capa/capa/responsetypes.py

```diff
@@ -1099,22 +1228,91 @@ def setup_response(self):
     def get_score(self, student_answers):
         # log.debug('%s: student_answers=%s' % (unicode(self),student_answers))
         cmap = CorrectMap()
+        
         amap = self.get_answers()
+        
+        tree = self.xml
+        problem_xml = tree.xpath('.')
+
+        # Partial credit type - can set 'points' only at the moment.
+        credit_type = problem_xml[0].get('partial_credit', default=False)
+
+        try:
+            credit_type = str(credit_type).lower().strip()
+        except ValueError:
+            _ = self.capa_system.i18n.ugettext
+            # Translators: 'partial_credit' is an attribute name and should not be translated.
+            msg = _("partial_credit value can only be set to 'points' or be removed.")
+            raise LoncapaProblemError(msg)
+        
+
         for aid in amap:
-            if aid in student_answers and student_answers[aid] == amap[aid]:
-                cmap.set(aid, 'correct')
-            else:
-                cmap.set(aid, 'incorrect')
+            for word in amap[aid]:
```

Comment:
Sorry to not be more clear. I have no idea what's going on here. So some comments in your code would really help, because I can't really review when I don't understand what's happening. As I said, I know you didn't write this code but you've taken the pain of figuring out how to use these poorly-named variables, so adding some comments would help others follow along.